### ISSUE TITLE
LR Scheduler instantiation fails when Optimizer loaded via DCP

### ISSUE THREAD

**Issue by le1nux** [Modalities/modalities#315]
The  lr scheduler instantiation with an optimizer that has been loaded via distributed checkpointing (DCP) fails, as the `inital_lr` has not been set by the lr_scheduler (in the case of `last_epoch != -1`). 

See here, for an extended discussion: 
https://github.com/pytorch/pytorch/issues/140900

Possible solution: 
Torch Titan also checkpoints the lr scheduler: 
https://github.com/pytorch/torchtitan/blob/main/torchtitan/components/checkpoint.py#L266

**Comment by le1nux** [comment]
implemented in #312
